### ISSUE TITLE
core:archive error: Label column not found in the table to add in addDataTable

### ISSUE THREAD

**Issue by catlan** [matomo-org/plugin-GoogleAnalyticsImporter#81]
Google Analytics Import

The archiver seems to have problem with the imported GA website. Every hour the crontab throws this error:

```
ERROR [2020-06-30 10:05:16] 7134  Got invalid response from API request: ?module=API&method=API.get&idSite=2&period=year&date=last3&format=php&trigger=archivephp. Response was 'a:2:{s:6:"result";s:5:"error";s:7:"message";s:21275:"Label column not found in the table to add in addDataTable() - in plugin Referrers #0 /var/www/virtual/XXXXX/analytics.XXXXX.com/core/ArchiveProcessor/Loader.php(165): Piwik\ArchiveProcessor\PluginsArchiver-&gt;callAggregateAllPlugins(1255, 0, false) matomo-org/matomo#1 /var/www/virtual/XXXXX/analytics.XXXXX.com/core/ArchiveProcessor/Loader.php(111): Piwik\ArchiveProcessor\Loader-&gt;prepareAllPluginsArchive(1255, 0) matomo-org/matomo#2 /var/www/virtual/XXXXX/analytics.XXXXX.com/core/ArchiveProcessor/Loader.php(84): Piwik\ArchiveProcessor\Loader-&gt;prepareArchiveImpl('VisitsSummary') matomo-org/matomo#3 /var/www/virtual/XXXXX/analytics.XXXXX.com/core/Context.php(75): Piwik\ArchiveProcessor\Loader-&gt;Piwik\ArchiveProcessor\{closure}() matomo-org/matomo#4 /var/www/virtual/XXXXX/analytics.XXXXX.com/core/ArchiveProcessor/Loader.php(85): Piwik\Context::changeIdSite(2, Object(Closure)) matomo-org/matomo#5 /var/www/virtual/XXXXX/analytics.XXXXX.com/core/Archive.php(803): Piwik\ArchiveProcessor\Loader-&gt;prepareArchive('VisitsSummary') matomo-org/matomo#6 /var/www/virtual/XXXXX/analytics.XXXXX.com/core/Archive.php(607): Piwik\Archive-&gt;prepareArchive(Array, Object(Piwik\Site), Object(Piwik\Period\Week)) matomo-org/matomo#7 /var/www/virtual/XXXXX/analytics.XXXXX.com/core/Archive.php(551): Piwik\Archive-&gt;cacheArchiveIdsAfterLaunching(Array, Array) matomo-org/matomo#8 /var/www/virtual/XXXXX/analytics.XXXXX.com/core/Archive.php(480): Piwik\Archive-&gt;getArchiveIds(Array) matomo-org/matomo#9 /var/www/virtual/XXXXX/analytics.XXXXX.com/core/Archive.php(295): Piwik\Archive-&gt;get(Array, 'numeric') matomo-org/matomo#10 /var/www/virtual/XXXXX/analytics.XXXXX.com/core/ArchiveProcessor.php(602): Piwik\Archive-&gt;getDataTableFromNumeric(Array) matomo-org/matomo#11 /var/www/virtual/XXXXX/analytics.XXXXX.com/core/ArchiveProcessor.php(247): Piwik\ArchiveProcessor-&gt;getAggregatedNumericMetrics(Array, false) matomo-org/matomo#12 /var/www/virtual/XXXXX/analytics.XXXXX.com/core/ArchiveProcessor/PluginsArchiver.php(307): Piwik\ArchiveProcessor-&gt;aggregateNumericMetrics(Array) matomo-org/matomo#13 /var/www/virtual/XXXXX/analytics.XXXXX.com/core/ArchiveProcessor/PluginsArchiver.php(108): Piwik\ArchiveProcessor\PluginsArchiver-&gt;aggregateMultipleVisitsMetrics() matomo-org/matomo#14 /var/www/virtual/XXXXX/analytics.XXXXX.com/core/ArchiveProcessor/Loader.php(159): Piwik\ArchiveProcessor\PluginsArchiver-&gt;callAggregateCoreMetrics() matomo-org/matomo#15 /var/www/virtual/XXXXX/analytics.XXXXX.com/core/ArchiveProcessor/Loader.php(111): Piwik\ArchiveProcessor\Loader-&gt;prepareAllPluginsArchive(0, 0) matomo-org/matomo#16 /var/www/virtual/XXXXX/analytics.XXXXX.com/core/ArchiveProcessor/Loader.php(84): Piwik\ArchiveProcessor\Loader-&gt;prepareArchiveImpl('VisitsSummary') matomo-org/matomo#17 /var/www/virtual/XXXXX/analytics.XXXXX.com/core/Context.php(75): Piwik\ArchiveProcessor\Loader-&gt;Piwik\ArchiveProcessor\{closure}() matomo-org/matomo#18 /var/www/virtual/XXXXX/analytics.XXXXX.com/core/ArchiveProcessor/Loader.php(85): Piwik\Context::changeIdSite(2, Object(Closure)) matomo-org/matomo#19 /var/www/virtual/XXXXX/analytics.XXXXX.com/core/Archive.php(803): Piwik\ArchiveProcessor\Loader-&gt;prepareArchive('VisitsSummary') matomo-org/matomo#20 /var/www/virtual/XXXXX/analytics.XXXXX.com/core/Archive.php(607): Piwik\Archive-&gt;prepareArchive(Array, Object(Piwik\Site), Object(Piwik\Period\Month)) matomo-org/matomo#21 /var/www/virtual/XXXXX/analytics.XXXXX.com/core/Archive.php(551): Piwik\Archive-&gt;cacheArchiveIdsAfterLaunching(Array, Array) matomo-org/matomo#22 /var/www/virtual/XXXXX/analytics.XXXXX.com/core/Archive.php(480): Piwik\Archive-&gt;getArchiveIds(Array) matomo-org/matomo#23 /var/www/virtual/XXXXX/analytics.XXXXX.com/core/Archive.php(295): Piwik\Archive-&gt;get(Array, 'numeric') matomo-org/matomo#24 /var/www/virtual/XXXXX/analytics.XXXXX.com/core/ArchiveProcessor.php(602): Piwik\Archive-&gt;getDataTableFromNumeric(Array) matomo-org/matomo#25 /var/www/virtual/XXXXX/analytics.XXXXX.com/core/ArchiveProcessor.php(247): Piwik\ArchiveProcessor-&gt;getAggregatedNumericMetrics(Array, false) matomo-org/matomo#26 /var/www/virtual/XXXXX/analytics.XXXXX.com/core/ArchiveProcessor/PluginsArchiver.php(307): Piwik\ArchiveProcessor-&gt;aggregateNumericMetrics(Array) matomo-org/matomo#27 /var/www/virtual/XXXXX/analytics.XXXXX.com/core/ArchiveProcessor/PluginsArchiver.php(108): Piwik\ArchiveProcessor\PluginsArchiver-&gt;aggregateMultipleVisitsMetrics() matomo-org/matomo#28 /var/www/virtual/XXXXX/analytics.XXXXX.com/core/ArchiveProcessor/Loader.php(159): Piwik\ArchiveProcessor\PluginsArchiver-&gt;callAggregateCoreMetrics() matomo-org/matomo#29 /var/www/virtual/XXXXX/analytics.XXXXX.com/core/ArchiveProcessor/Loader.php(111): Piwik\ArchiveProcessor\Loader-&gt;prepareAllPluginsArchive('24900', '2') matomo-org/matomo#30 /var/www/virtual/XXXXX/analytics.XXXXX.com/core/ArchiveProcessor/Loader.php(84): Piwik\ArchiveProcessor\Loader-&gt;prepareArchiveImpl('VisitsSummary') matomo-org/matomo#31 /var/www/virtual/XXXXX/analytics.XXXXX.com/core/Context.php(75): Piwik\ArchiveProcessor\Loader-&gt;Piwik\ArchiveProcessor\{closure}() matomo-org/matomo#32 /var/www/virtual/XXXXX/analytics.XXXXX.com/core/ArchiveProcessor/Loader.php(85): Piwik\Context::changeIdSite(2, Object(Closure)) matomo-org/matomo#33 /var/www/virtual/XXXXX/analytics.XXXXX.com/core/Archive.php(803): Piwik\ArchiveProcessor\Loader-&gt;prepareArchive('VisitsSummary') matomo-org/matomo#34 /var/www/virtual/XXXXX/analytics.XXXXX.com/core/Archive.php(607): Piwik\Archive-&gt;prepareArchive(Array, Object(Piwik\Site), Object(Piwik\Period\Year)) matomo-org/matomo#35 /var/www/virtual/XXXXX/analytics.XXXXX.com/core/Archive.php(551): Piwik\Archive-&gt;cacheArchiveIdsAfterLaunching(Array, Array) matomo-org/matomo#36 /var/www/virtual/XXXXX/analytics.XXXXX.com/core/Archive.php(480): Piwik\Archive-&gt;getArchiveIds(Array) matomo-org/matomo#37 /var/www/virtual/XXXXX/analytics.XXXXX.com/core/Archive.php(295): Piwik\Archive-&gt;get(Array, 'numeric') matomo-org/matomo#38 /var/www/virtual/XXXXX/analytics.XXXXX.com/plugins/VisitsSummary/API.php(36): Piwik\Archive-&gt;getDataTableFromNum ... alytics.XXXXX.com/console(32): Symfony\Component\Console\Application-&gt;run() matomo-org/matomo#77 {main}";}'
ERROR [2020-06-30 10:05:17] 7134  1 total errors during this script execution, please investigate and try and fix these errors.
ERROR [2020-06-30 10:05:17] 7134  Uncaught exception: /var/www/virtual/XXXXX/analytics.XXXXX.com/core/CronArchive.php(606): 1 total errors during this script execution, please investigate and try and fix these errors.
1 total errors during this script execution, please investigate and try and fix these errors.
#0 /var/www/virtual/XXXXX/analytics.XXXXX.com/core/CronArchive.php(599): Piwik\CronArchive->logFatalError('1 total errors ...')
matomo-org/matomo#1 /var/www/virtual/XXXXX/analytics.XXXXX.com/core/CronArchive.php(331): Piwik\CronArchive->end()
matomo-org/matomo#2 /var/www/virtual/XXXXX/analytics.XXXXX.com/core/Access.php(635): Piwik\CronArchive->Piwik\{closure}()
matomo-org/matomo#3 /var/www/virtual/XXXXX/analytics.XXXXX.com/core/CronArchive.php(335): Piwik\Access::doAsSuperUser(Object(Closure))
matomo-org/matomo#4 /var/www/virtual/XXXXX/analytics.XXXXX.com/plugins/CoreConsole/Commands/CoreArchiver.php(27): Piwik\CronArchive->main()
matomo-org/matomo#5 /var/www/virtual/XXXXX/analytics.XXXXX.com/vendor/symfony/console/Symfony/Component/Console/Command/Command.php(257): Piwik\Plugins\CoreConsole\Commands\CoreArchiver->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
matomo-org/matomo#6 /var/www/virtual/XXXXX/analytics.XXXXX.com/vendor/symfony/console/Symfony/Component/Console/Application.php(874): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
matomo-org/matomo#7 /var/www/virtual/XXXXX/analytics.XXXXX.com/vendor/symfony/console/Symfony/Component/Console/Application.php(195): Symfony\Component\Console\Application->doRunCommand(Object(Piwik\Plugins\CoreConsole\Commands\CoreArchiver), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
matomo-org/matomo#8 [internal function]: Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
matomo-org/matomo#9 /var/www/virtual/XXXXX/analytics.XXXXX.com/core/Console.php(140): call_user_func(Array, Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
matomo-org/matomo#10 /var/www/virtual/XXXXX/analytics.XXXXX.com/core/Access.php(644): Piwik\Console->Piwik\{closure}()
matomo-org/matomo#11 /var/www/virtual/XXXXX/analytics.XXXXX.com/core/Console.php(141): Piwik\Access::doAsSuperUser(Object(Closure))
matomo-org/matomo#12 /var/www/virtual/XXXXX/analytics.XXXXX.com/core/Console.php(93): Piwik\Console->doRunImpl(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
matomo-org/matomo#13 /var/www/virtual/XXXXX/analytics.XXXXX.com/vendor/symfony/console/Symfony/Component/Console/Application.php(126): Piwik\Console->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
matomo-org/matomo#14 /var/www/virtual/XXXXX/analytics.XXXXX.com/console(32): Symfony\Component\Console\Application->run()
matomo-org/matomo#15 {main}



 [Exception]                                                                                    
 1 total errors during this script execution, please investigate and try and fix these errors.  



core:archive [--url="..."] [--force-all-websites] [--force-all-periods[="..."]] [--force-timeout-for-periods[="..."]] [--skip-idsites[="..."]] [--skip-all-segments] [--force-idsites[="..."]] [--skip-segments-today] [--force-periods[="..."]] [--force-date-last-n="..."] [--force-date-range[="..."]] [--force-idsegments="..."] [--concurrent-requests-per-website[="..."]] [--concurrent-archivers[="..."]] [--disable-scheduled-tasks] [--accept-invalid-ssl-certificate] [--php-cli-options[="..."]]
```

Another report of the same issue can be found here: https://forum.matomo.org/t/label-column-not-found-in-the-table-to-add-in-adddatatable/36491

**Comment by diosmosis** [comment]
This occurs due to the data in GA, the easiest way for me to debug this would be if you could provide access to your GA account so I can reproduce the issue. If this is possible, please email `support@matomo.org` with the oauth credentials for access. If not it will take longer to find the root cause. And can you let me know what version of the plugin you are using?

**Comment by catlan** [comment]
GoogleAnalyticsImporter (v1.4.0)

Also followed up with support@matomo.org regarding GA access.

**Comment by diosmosis** [comment]
Fixed by #83
